integr8ly/application-monitoring-operator
# Full installation

Implements the installation of the monitoring stack by deploying the Grafana and Prometheus Operators and configuring all resources. You should end up with a full monitoring stack including an alert manager and a service monitor that is capable of monitoring itself.

It also creates a route to the AlertManager which you can find under `Application -> Routes`

Some of the resources can't be created by the operator because they are cluster level. This is handled by the installer, see: https://github.com/integr8ly/installation/pull/311

Verification steps:

You can run this locally against a cluster, but you must create some of the resources first:

* Create a namespace named `middleware-monitoring`
* Create the cluster level resources: `oc apply -f deploy/roles`
* Create the operator roles: `oc apply -f deploy/operator_roles`
* Create the CRDs: `oc apply -f deploy/crds`
* Make sure the CRDs of the Grafana Operator are installed: `oc apply -f <path to grafana operator>/deploy/crds`
* Run the operator locally: `make code/run`
* Deploy the monitoring stack: `oc create -f deploy/examples/ApplicationMonitoring.yaml`
